tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -54,10 +54,66 @@
     fake_download_file,
     fake_magpie_api_request,
 )
-from utils import get_halted_workflow
 from inspirehep.modules.workflows.tasks.matching import _get_hep_record_brief
 
 
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link'
+)
+def get_halted_workflow(mocked_is_pdf_link, app, record, extra_config=None):
+    mocked_is_pdf_link.return_value = True
+
+    extra_config = extra_config or {}
+    with mock.patch.dict(app.config, extra_config):
+        workflow_uuid = start('article', [record])
+
+    eng = WorkflowEngine.from_uuid(workflow_uuid)
+    obj = eng.processed_objects[0]
+
+    assert obj.status == ObjectStatus.HALTED
+    assert obj.data_type == "hep"
+
+    # Files should have been attached (tarball + pdf, and plots)
+    assert obj.files["1407.7587.pdf"]
+    assert obj.files["1407.7587.tar.gz"]
+
+    assert len(obj.files) > 2
+
+    # A publication note should have been extracted
+    pub_info = obj.data.get('publication_info')
+    assert pub_info
+    assert pub_info[0]
+    assert pub_info[0].get('year') == 2014
+    assert pub_info[0].get('journal_title') == "J. Math. Phys."
+
+    # A prediction should have been made
+    prediction = obj.extra_data.get("relevance_prediction")
+    assert prediction
+    assert prediction['decision'] == 'Non-CORE'
+    assert prediction['scores']['Non-CORE'] == 0.8358207729691823
```

Comment:
I'm not sure if it's the case, but could comparing floating-point numbers for equality here get flaky?